src/lib/vip-import-validate-files.js

```diff
@@ -187,80 +199,84 @@ export const findNestedDirectories = async directory => {
-	const directories = folderStructure.split( '/' );
-
-	/**
-	 * Upload folder validation
-	 *
-	 * Find if an `uploads` folder exists and return its index position
-	 */
-	const uploadsIndex = directories.indexOf( 'uploads' );
-
-	/**
-	 * Year folder validation
-	 *
-	 * Find if a year folder exists via a four digit regex matching pattern,
-	 * then obtain that value
-	 */
-	const regexYear = /\b\d{4}\b/g;
-	const year = regexYear.exec( folderStructure ); // Returns an array with the regex-matching value
-
-	if ( year ) {
-		yearIndex = directories.indexOf( year[ 0 ] );
-	}
 
-	/**
-	 * Month folder validation
-	 *
-	 * Find if a month folder exists via a two digit regex matching pattern,
-	 * then obtain that value
-	 */
-	const regexMonth = /\b\d{2}\b/g;
-	const month = regexMonth.exec( folderStructure ); // Returns an array with the regex-matching value
-
-	if ( month ) {
-		monthIndex = directories.indexOf( month[ 0 ] );
-	}
+	// Loop through each key (path) to validate the folder structure format
+	for ( const folderPath of folderStructureKeys ) {
+		let yearIndex, monthIndex;
+
+		console.log( chalk.bold( 'Folder:' ), chalk.cyan( `${ folderPath }` ) );
+
+		// Turn the path into an array to determine index position
+		const directories = folderPath.split( '/' );
+
+		/**
+			* Upload folder validation
+			*
+			* Find if an `uploads` folder exists and return its index position
+			*/
+		const uploadsIndex = directories.indexOf( 'uploads' );
+
+		/**
+			* Year folder validation
+			*
+			* Find if a year folder exists via a four digit regex matching pattern,
+			* then obtain that value
+			*/
+		const regexYear = /\b\d{4}\b/g;
+		const year = regexYear.exec( folderPath ); // Returns an array with the regex-matching value
+
+		if ( year ) {
+			yearIndex = directories.indexOf( year[ 0 ] );
+		}
 
-	/**
-	 * Logging
-	 */
-
-	// Uploads folder
-	if ( uploadsIndex === 0 ) {
-		console.log();
-		console.log( '✅ File structure: Uploads directory exists' );
-		console.log();
-	} else {
-		console.log();
-		console.log( chalk.yellow( '✕' ), 'Recommended: Media files should reside in an', chalk.magenta( '`uploads`' ), 'directory' );
-		errors++;
-	}
+		/**
+			* Month folder validation
+			*
+			* Find if a month folder exists via a two digit regex matching pattern,
+			* then obtain that value
+			*/
+		const regexMonth = /\b\d{2}\b/g;
+		const month = regexMonth.exec( folderPath ); // Returns an array with the regex-matching value
+
+		if ( month ) {
+			monthIndex = directories.indexOf( month[ 0 ] );
+		}
 
-	// Year folder
-	if ( yearIndex && yearIndex === 1 ) {
-		console.log( '✅ File structure: Year directory exists (format: YYYY)' );
-		console.log();
-	} else {
-		console.log( chalk.yellow( '✕' ), 'Recommended: Structure your WordPress media files into', chalk.magenta( '`uploads/YYYY`' ), 'directories' );
-		errors++;
-	}
+		/**
+			* Logging
+			*/
+
+		// Uploads folder
+		if ( uploadsIndex === 0 ) {
```

Comment:
I think the index parts are what didn't pass for me if I supply a path argument that isn't relative to the test execution (ie. `~/Downloads/upload`):

![image](https://user-images.githubusercontent.com/1752747/90286501-061e1a80-de44-11ea-8026-36411a543635.png)

We might have to re-evaluate how we reference this.  Maybe we can skip validating what the index is of the uploads directory and just ensure that the year and month are at the upload directory index +1 and +2 respectively?  That's one way, but perhaps there's another if we have to validate the upload directory position?

Reply:
Thank you for the feedback 🙏 ! I think in this instance, it's working as expected- if the supplied media folder is `Downloads/uploads`, the recommended structure is simply an `uploads` folder (without the `Downloads` folder). Maybe the wording needs to be clearer?